src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -69,10 +73,20 @@ public int getMinWeightMagnitude() {
         return MIN_WEIGHT_MAGNITUDE;
     }
 
-    private static void runValidation(TransactionViewModel transactionViewModel, final int minWeightMagnitude) {
+    private static boolean invalidTimestamp(TransactionViewModel transactionViewModel) {
+        if (transactionViewModel.getAttachmentTimestamp() == 0) {
+            return transactionViewModel.getTimestamp() < MIN_TIMESTAMP && !transactionViewModel.getHash().equals(Hash.NULL_HASH)
```

Comment:
For defensive programming it is better to use `Objects.equals(transactionViewModel.getHash(), Hash.NULL_HASH)` or at least `Hash.NULL_HASH.equals(transaction.getHash())`.

Unless you WANT an exception to be thrown if `transactionViewModel.getHash() == null`, of course (then it will be nice to precede with `Objects.requireNonNull()`.

Even if you claim `transactionViewModel.getHash() == null` is an impossibility consider the following:
1. Man throughout history proved that the impossible is possible.
2. Even if it is impossible, getting into the habit of using `Objects.equals()` in general eliminates pesky null pointer exceptions without even thinking about them. So I want us to adopt that habit.